src/main/java/org/mozilla/zest/core/v1/ZestRequest.java

```diff
@@ -77,11 +78,11 @@ public ZestRequest deepCopy () {
 			zr.addAssertion((ZestAssertion)zt.deepCopy());
 		}
 		
-		for (Cookie cookie : this.cookies) {
-			zr.addCookie(new Cookie(cookie.getDomain(), cookie.getName(), cookie.getValue(),
+		for (ZestCookie cookie : this.cookies) {
+			zr.addCookie(new ZestCookie(cookie.getDomain(), cookie.getName(), cookie.getValue(),
 					cookie.getPath(), cookie.getExpiryDate(), cookie.getSecure()));
 		}
-		zr.cookies = this.cookies;
```

Comment:
- This warrants a test, hopefully the current code is not relying on this bug. 
 - Ideally these behaviour changes should be done in another changeset than a refactor (specially if not noted anywhere), makes it easer to review.